conda_lock/conda_lock.py

```diff
@@ -410,7 +428,9 @@ def make_lock_files(
                     file=sys.stderr,
                 )
 
-        assert lock_content is not None
+        # After this point, we're working with `new_lock_content`, never
+        # `original_lock_content` or `fresh_lock_content`.
+        assert new_lock_content is not None
```

Comment:
I found this assertion a bit confusing. I was not able to type `new_lock_content: Lockfile` because `original_lock_content` could be `None` when we assign it to `new_lock_content` in the `if not platforms_to_lock` block.

I would like to move this check in to the type system, but that seems better left for a larger refactor effort.